src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/custom.py

```diff
@@ -0,0 +1,167 @@
+        from azure.cli.core._profile import Profile, CLOUD
+        profile = Profile()
+        cred, _, tenant_id = profile.get_login_credentials(
+            resource=CLOUD.endpoints.active_directory_graph_resource_id)
+        graph_client = GraphRbacManagementClient(cred,
+                                                 tenant_id,
+                                                 base_url=CLOUD.endpoints.active_directory_graph_resource_id)
+        subscription = profile.get_subscription()
+        object_id = _get_object_id(graph_client, subscription=subscription)
+        odata_filter = "Properties/ownerObjectId eq '{}'".format(object_id)
+    elif my_claimable_vms:
+        odata_filter = 'properties/allowClaim'
+    else:
+        odata_filter = filters
+
+    return client.list(resource_group, lab_name, odata_filter, top, order_by)
+
+# TODO: Following methods can be extracted into common utils shared by other command modules
+
+
+def _single(collection):
+    return len([x for x in collection if x]) == 1
+
+
+def _get_object_id(graph_client, subscription=None, spn=None, upn=None):
+    if spn:
+        return _get_object_id_by_spn(graph_client, spn)
+    if upn:
+        return _get_object_id_by_upn(graph_client, upn)
+    return _get_object_id_from_subscription(graph_client, subscription)
+
+
+def _get_object_id_from_subscription(graph_client, subscription):
+    if subscription['user']:
+        if subscription['user']['type'] == 'user':
+            return _get_object_id_by_upn(graph_client, subscription['user']['name'])
+        elif subscription['user']['type'] == 'servicePrincipal':
+            return _get_object_id_by_spn(graph_client, subscription['user']['name'])
+        else:
+            logger.warning("Unknown user type '%s'", subscription['user']['type'])
+    else:
+        logger.warning('Current credentials are not from a user or service principal. '
+                       'Azure Key Vault does not work with certificate credentials.')
+
+
+def _get_object_id_by_spn(graph_client, spn):
+    accounts = list(graph_client.service_principals.list(
+        filter="servicePrincipalNames/any(c:c eq '{}')".format(spn)))
+    if not accounts:
+        logger.warning("Unable to find user with spn '%s'", spn)
+        return
+    if len(accounts) > 1:
+        logger.warning("Multiple service principals found with spn '%s'. "
+                       "You can avoid this by specifying object id.", spn)
+        return
+    return accounts[0].object_id
+
+
+def _get_object_id_by_upn(graph_client, upn):
+    accounts = list(graph_client.users.list(filter="userPrincipalName eq '{}'".format(upn)))
+    if not accounts:
+        logger.warning("Unable to find user with upn '%s'", upn)
+        return
+    if len(accounts) > 1:
+        logger.warning("Multiple users principals found with upn '%s'. "
+                       "You can avoid this by specifying object id.", upn)
+        return
+    return accounts[0].object_id
```

Comment:
Yup. You are right. I have marked the set of methods that i am reusing into the common utils once we finalize the code in separate PR.